third_party/move/move-compiler-v2/src/env_pipeline/inliner.rs

```diff
@@ -710,11 +820,51 @@ impl<'env, 'rewriter> InlinedRewriter<'env, 'rewriter> {
 
         let lambda_param_map: BTreeMap<Symbol, &Exp> = lambda_args_matched
             .iter()
-            .map(|(param, arg_exp)| (param.0, *arg_exp))
+            .map(|(param, arg_exp)| (param.1 .0, *arg_exp))
             .collect();
 
-        let (regular_params, regular_actuals): (Vec<&Parameter>, Vec<&Exp>) =
+        let mut function_value_map: BTreeMap<usize, Exp> = BTreeMap::new();
+        let mut lifted_lambda_funs: BTreeMap<usize, move_model::model::FunctionData> =
```

Comment:
Should we move this for clarity in its own function?

Reply:
moved into a standalone function.